docs/source/configuration.rst

```diff
@@ -0,0 +1,388 @@
+=======================
+Configuration Reference
+=======================
+
+This page documents the schema of the YAML configuration file used by the Migrator and the Validator.
+
+The configuration file is a YAML object whose fields are enumerated, preceded by a comment describing their role. All the fields are mandatory except if their documentation starts with “Optional”. All the values of the form ``<xxx>`` are placeholders that should be replaced with your specific settings.
+
+The configuration file requires the following top-level properties, which are documented further below:
+
+.. code-block:: yaml
+
+  # Source configuration
+  source:
+    # ...
+  # Target configuration
+  target:
+    # ...
+  # Columns to rename
+  renames:
+    # ...
+  # Savepoints configuration
+  savepoints:
+    # ...
+  # Validator configuration
+  validation:
+    # ...
+  # Used internally
+  skipTokenRanges: []
+
+These top-level properties are documented in the following sections (except ``skipTokenRanges``, which is used internally).
+
```

Comment:
Maybe we should provide a sample of error when something is missing or there is a typo (this DropDown ... cryptic error the config file parser prints)

Decoding that error is the second most seen user error - they don't know what to fix and where

Reply:
oh, fixed in circe 0.14.7
but let's add at least sample if something is missing, e.g.

```
io.circe.DecodingFailure$DecodingFailureImpl: DecodingFailure at .skipTokenRanges: Missing required field
```
so people can find such config errors using google

Reply:
Fixed in [4da9711](https://github.com/scylladb/scylla-migrator/pull/166/commits/4da971193facbee455c49626955729ec21a54a4d).